.cirrus.yml

```diff
@@ -1,547 +1,499 @@
-    - bin/cache/dart-sdk/bin/dart --enable-asserts dev/bots/test.dart
-  matrix:
-    # all of the tests except test/integration and test/commands/create_test for packages/flutter_tools
-    - name: tool_tests-macos
-      env:
-        GCLOUD_SERVICE_ACCOUNT_KEY: ENCRYPTED[f12abe60f5045d619ef4c79b83dd1e0722a0b0b13dbea95fbe334e2db7fffbcd841a5a92da8824848b539a19afe0c9fb]
-        SHARD: tool_tests
-        SUBSHARD: tool
-        SHARD_INDEX: 1
-    # all of the tests in test/commands/create_test
-    - name: tool_tests_create-macos
-      env:
-        GCLOUD_SERVICE_ACCOUNT_KEY: ENCRYPTED[f12abe60f5045d619ef4c79b83dd1e0722a0b0b13dbea95fbe334e2db7fffbcd841a5a92da8824848b539a19afe0c9fb]
-        SHARD: tool_tests
-        SUBSHARD: create
-    # all of the tests in test/integration for packages/flutter_tools
-    - name: tool_tests_integration-macos
-      env:
-        GCLOUD_SERVICE_ACCOUNT_KEY: ENCRYPTED[f12abe60f5045d619ef4c79b83dd1e0722a0b0b13dbea95fbe334e2db7fffbcd841a5a92da8824848b539a19afe0c9fb]
-        SHARD: tool_tests
-        SUBSHARD: integration
-
-task:
-  use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true'
-  osx_instance:
-    image: mojave-xcode-10.1
-  depends_on:
-    - analyze
-  env:
-    CIRRUS_WORKING_DIR: "/tmp/flutter sdk"
-    COCOAPODS_DISABLE_STATS: true
-    PATH: "$CIRRUS_WORKING_DIR/bin:$CIRRUS_WORKING_DIR/bin/cache/dart-sdk/bin:$PATH"
-  # occasionally the clock on these machines is out of sync
-  # with the actual time - this should help to verify
-  print_date_script:
     - date
-  install_cocoapods_script:
+    - which flutter
     - sudo gem install cocoapods
-  git_fetch_script:
-    - git clean -xfd
+    - git clean -xffd
     - git fetch origin
-    - git fetch origin master # To set FETCH_HEAD for "git merge-base" to work
-  pub_cache:
-    folder: $HOME/.pub-cache
-    fingerprint_script: echo $OS; grep -r --include=pubspec.yaml 'PUBSPEC CHECKSUM' "$CIRRUS_WORKING_DIR"
-  flutter_pkg_cache:
-    folder: bin/cache/pkg
-    fingerprint_script: echo $OS; cat bin/internal/engine.version
-  artifacts_cache:
-    folder: bin/cache/artifacts
-    fingerprint_script: echo $OS; cat bin/internal/engine.version
-  setup_script:
-    - bin/flutter config --no-analytics
-    - bin/flutter doctor -v
-    - bin/flutter update-packages
+    - git fetch origin master # To set FETCH_HEAD, so that "git merge-base" works.
+    - flutter config --no-analytics
+    - flutter doctor -v
+    - flutter update-packages
+    - date
+    - which flutter
   matrix:
-    - name: tests_widgets-macos
+    - name: tests_framework_widgets-macos
       env:
-        GCLOUD_SERVICE_ACCOUNT_KEY: ENCRYPTED[f12abe60f5045d619ef4c79b83dd1e0722a0b0b13dbea95fbe334e2db7fffbcd841a5a92da8824848b539a19afe0c9fb]
-        SHARD: tests
+        SHARD: framework_tests
         SUBSHARD: widgets
-      test_all_script:
+      script:
         - ulimit -S -n 2048 # https://github.com/flutter/flutter/issues/2976
```

Comment:
Why don't we just move the `ulimit -S -n 2048` up to the setup step?  There's no harm in having it for all shards, and it can only hurt us to not have it.

Reply:
Actually, we could just move the dart part up there too - the env part of the matrix will capture the differences.

Reply:
I guess that would require changing customer testint though, so maybe not.  But the ulimit part can move.

Reply:
different scripts run in different shells so moving it up doesn't work.

Reply:
Ahh and ulimit is a builtin.  Shame.